system/hardware/tici/agnos.py

```diff
@@ -172,9 +165,68 @@ def flash_partition(target_slot_number: int, partition: dict, cloudlog):
     if out.tell() != partition['size']:
       raise Exception("Uncompressed size mismatch")
 
-    # Write hash after successfull flash
     os.sync()
-    if not full_check:
+
+
+def extract_casync_image(target_slot_number: int, partition: dict, cloudlog):
+  path = get_partition_path(target_slot_number, partition)
+  seed_path = path[:-1] + ('b' if path[-1] == 'a' else 'a')
+
+  target = casync.parse_caibx(partition['casync_caibx'])
+
+  sources = []
+
+  # First source is the current partition. Index file for current version is provided in the manifest
+  # TODO: build url based on agnos version or parition hash instead?
+  if 'casync_seed_caibx' in partition:
+    sources += [('seed', functools.partial(casync.read_chunk_local_file, f=open(seed_path, 'rb')), casync.build_chunk_dict(casync.parse_caibx(partition['casync_seed_caibx'])))]
```

Comment:
we should be better about using context manager or explicitly closing files. this is going to be a problem once we start setting PYTHONWARNINGS=error in CI